v0/destinations/yahoo_dsp/transform.js

```diff
@@ -0,0 +1,260 @@
+    "42cbe7eebb412bbcd5b56",
+    "379b4653a40878da7a584"
+  ]
+ */
+
+const populateIdentifiers = (attributeArray, { Config }) => {
+  const seedList = [];
+  const { audienceType } = Config;
+  const { hashRequired } = Config;
+  let listType;
+  if (isDefinedAndNotNullAndNotEmpty(attributeArray)) {
+    // traversing through every element in the add array for the elements to be added.
+    attributeArray.forEach(element => {
+      // storing keys of an object inside the add array.
+      const keys = Object.keys(element);
+      // checking for the audience type the user wants to add is present in the input or not.
+      keys.forEach(key => {
+        if (key === audienceType) {
+          listType = audienceType;
+        }
+      });
+      // throwing error if the audience type the user wants to add is not present in the input.
+      if (!listType) {
+        throw new CustomError(
+          `Audience type ${audienceType} not provided`,
+          400
+        );
+      }
+      // here, hashing the data if is not hashed and pushing in the seedList array.
+      if (hashRequired) {
+        seedList.push(sha256(element[audienceType]));
+      } else {
+        seedList.push(element[audienceType]);
+      }
+    });
+  }
+  return seedList;
+};
+
+/**
+ * This function is used for building the final response to be returned.
+ * @param {*} message
+ * @param {*} destination
+ * @returns
+ */
+
+const responseBuilder = async (message, destination) => {
+  const { listData } = message.properties;
+  const {
+    accountId,
+    audienceId,
+    audienceType,
+    seedListType
+  } = destination.Config;
+
+  let listType;
+  let outputPayload = {};
+  // The only supported property for now is "add"
+  const key = "add";
+
+  const domains = [];
+  const categoryIds = [];
+  let seedList = [];
+
+  /**
+   * The below written switch case is used to build the response for each of the supported audience type.
+   *  eg. ["email", "deviceId", "ipAddress", "mailDomain", "pointOfInterest"].
+   */
+  switch (audienceType) {
+    case "email":
+      // seedList contains the list of emails to be updated
+      seedList = populateIdentifiers(listData[key], destination);
+      if (seedList.length === 0) {
+        throw new CustomError(
+          `[Yahoo_DSP]:: No attributes are present in the '${key}' property.`,
+          400
+        );
+      }
+      outputPayload = { ...outputPayload, accountId, seedList };
+      break;
+    case "deviceId":
+      // seedList contains the list of deviceIds to be updated
+      seedList = populateIdentifiers(listData[key], destination);
+      // throwing the error if nothing is present in the seedList
+      if (seedList.length === 0) {
+        throw new CustomError(
+          `[Yahoo_DSP]:: No attributes are present in the '${key}' property.`,
+          400
+        );
+      }
+      // throwing error if seedListType is not provided for deviceId type audience
+      if (!seedListType) {
```

Comment:
This seedListType should be verified first as it has only two possible values viz. `IDFA` and `GPADVID`.